core/account/indexer.go

```diff
@@ -22,33 +24,41 @@ const PinName = "account"
 // If the Core is configured not to provide search services,
 // SaveAnnotatedAccount can be a no-op.
 type Saver interface {
-	SaveAnnotatedAccount(context.Context, string, map[string]interface{}) error
+	SaveAnnotatedAccount(context.Context, *query.AnnotatedAccount) error
 }
 
 func (m *Manager) indexAnnotatedAccount(ctx context.Context, a *Account) error {
 	if m.indexer == nil {
 		return nil
 	}
-	var keys []map[string]interface{}
+
+	aa := &query.AnnotatedAccount{
```

Comment:
Maybe the part of this that creates the AnnotatedAccount could be extracted and used in core as well? Same for AnnotatedAsset?